### DISCUSSION TITLE
npx prisma generate, can I tell Prisma to use existing libssl.so.10 instead of libssl.so.1.1?

### DISCUSSION THREAD

**Discussion by namleasia** [prisma/prisma#8336] [Q&A]
I build a nestjs project with prisma, and upload to shared cpanel hosting, then run this command:
`npx prisma generate`

and get this error:
![image](https://user-images.githubusercontent.com/37680080/126057502-d2dac86f-5775-4313-af15-20c813e4dcec.png)

My ssl:
![image](https://user-images.githubusercontent.com/37680080/126064927-213cb54b-e6af-4ed8-94a4-5668d40a5dec.png)

How to fix this, please help!
Can I tell prisma to use libssl.so.10 instead of libssl.so.1.1?

**Comment by pantharshit00** [comment]
Can you please tell which operating system this box is using? Prisma links openssl as a dynamic library as it is required for secure connection to the database. 

Run `openssl version` and paste the output here. From the screenshot, it looks like you have version `1.0.x` so maybe try forcing Prisma to download openssl 1.0.x compatible binary by adding the following in your schema:
```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.0.x"] // or rhel-openssl-1.0.x if it is a redhat based system
}
```

Run `prisma generate` after this.

**Reply by namleasia** [reply]
openssl version:
OpenSSL 1.1.1b FIPS  26 Feb 2019

**Reply by ryands17** [reply]
@namleasia 👋 

Did the above work for you?

**Reply by pantharshit00** [reply]
Try the following since you have openssl 1.1
```prisma
generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "debian-openssl-1.1.x"] // or rhel-openssl-1.1.x if it is a redhat based system
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
